spark/src/main/java/org/apache/iceberg/spark/source/Writer.java

```diff
@@ -303,96 +305,179 @@ public String toString() {
         }
       }
     }
+
+    private class EncryptedOutputFileFactory implements OutputFileFactory<EncryptedOutputFile> {
```

Comment:
I don't think `OutputFileFactory` is needed. This is the only implementation of it. This could also be named `OutputFileFactory` and not mention encryption because the files may not actually be encrypted if the plaintext encryption manager is used.

Reply:
Agreed.